lib/edgehog_device_forwarder_web/device_socket.ex

```diff
@@ -0,0 +1,90 @@
+# Copyright 2023 SECO Mind Srl
+# SPDX-License-Identifier: Apache-2.0
+
+defmodule EdgehogDeviceForwarderWeb.DeviceSocket do
+  @moduledoc """
+  Socket implementation for the Device-Server communication.
+  """
+
+  @behaviour Phoenix.Socket.Transport
+
+  alias Plug.Conn
+  alias EdgehogDeviceForwarder.Forwarder
+  alias EdgehogDeviceForwarder.Tokens
+  alias EdgehogDeviceForwarder.WebSockets
+  require Logger
+
+  @type state :: Phoenix.Socket.Transport.state()
+
+  @spec child_spec(keyword) :: :ignore
+  def child_spec(_opts) do
+    :ignore
+  end
```

Comment:
Since it's the first time I see this used, can you detail why is this needed?

Reply:
`child_spec/1` is required from the `Phoenix.Socket.Transport` specification.
From the [documentation](https://hexdocs.pm/phoenix/Phoenix.Socket.Transport.html#c:child_spec/1), 

> `:ignore` means no child spec is necessary for this socket

Which is the case for us, as we don't spawn any additional process from the socket